src/main/kotlin/io/sip3/commons/domain/payload/ByteArrayPayload.kt

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2018-2019 SIP3.IO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.sip3.commons.domain.payload
+
+import io.netty.buffer.ByteBuf
+import io.netty.buffer.Unpooled
+
+class ByteArrayPayload() : Payload {
+
+    lateinit var bytes: ByteArray
+
+    constructor(bytes: ByteArray) : this() {
+        this.bytes = bytes
+    }
+
+    override fun encode(): ByteBuf {
+        return Unpooled.wrappedBuffer(bytes)
+    }
+
+    override fun decode(byteBuf: ByteBuf) {
+        bytes = byteBuf.array()
```

Comment:
better to use `byteBuf.getBytes()` because `byteBuf.array()` will throw an exception for `DirectByteBuffer`